delocate/libsana.py

```diff
@@ -104,6 +113,63 @@ def resolve_rpath(lib_path, rpaths):
     return lib_path
 
 
+def search_environment_for_lib(lib_path):
```

Comment:
And this differs from `resolve_rpath` because we can use the basename to search for the library in the specified paths?  So we can find a library specified as`/foo/bar/baz.dylib` at `/path/baz.dylib` where `/path` is in, say, `DYLIB_LIBRARY_PATH`?  Please forgive my slowness.

Reply:
Just so -- in your example, if `/foo/bar/baz.dylib` exists, and there is also a second `baz.dylib` at `/path/baz.dylib`, then by specifying `DYLD_LIBRARY_PATH=/path` you'll find the `/path` library before the `/foo/bar` one. Check out `test_dyld_library_path_beats_basename()` below.

Reply:
Sorry - just for my reassurance - if required install name is a full absolute path, such as `/foo/bar/baz.dylib`, rather than a relative path, `baz.dylib`, macOS will nevertheless find `/path/baz.dylib` (where `/path` is in `DYLD_LIBRARY_PATH`) before `/foo/bar/baz.dylib`?

Reply:
Let me go check that specifically -- I can add it to the test script afterwards (assuming it does in fact work as you describe, which it should!).

Reply:
Right, I've updated `testing_osx_rpath_env_variables.sh` with a demonstration of this working just like you describe. Give it a go. =)